care/abdm/utils/api_call.py

```diff
@@ -676,7 +677,18 @@ def on_data_request(self, data):
         return response
 
     def data_transfer(self, data):
-        headers = {"Content-Type": "application/json"}
+        token = cache.get(ABDM_TOKEN_CACHE_KEY)
+
+        if not token:
+            Request("").auth_header()
+            token = cache.get(ABDM_TOKEN_CACHE_KEY)
+            if not token:
+                return None
+
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": "Bearer " + cache.get(ABDM_TOKEN_CACHE_KEY),
+        }
```

Comment:
```suggestion
        auth_header = Request("").auth_header()
        
        if not auth_header:
            return None

        headers = {
            "Content-Type": "application/json",
            **auth_header
        }
```
Request("").auth_header() seems to be checking the cache already, so we can remove the duplicate logic